bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableVersionInfo.java

```diff
@@ -0,0 +1,53 @@
+package com.google.cloud.bigtable.config;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+public class BigtableVersionInfo {
+
+  private static final Logger LOG = new Logger(BigtableVersionInfo.class);
+
+  public static final String CLIENT_VERSION = getVersion();
+  public static final String JDK_VERSION = getJavaVersion();
+
+  public static final String CORE_UESR_AGENT = "bigtable-" + CLIENT_VERSION +",jdk-" + JDK_VERSION;
+
+  /**
+   * Gets user agent from bigtable-hbase.properties. Returns a default dev user agent with current
+   * timestamp if not found.
+   */
+  private static String getVersion() {
+    final String defaultVersion = "dev-" + System.currentTimeMillis();
```

Comment:
We can do it in a follow-up change but I'd strongly recommend using the ident gitattributes facility so we always embed the git commit hash in the user agent: 

https://git-scm.com/docs/gitattributes

That should be fairly straightforward to do.